service/src/main/java/uk/gov/hmcts/reform/fpl/service/ManageDocumentService.java

```diff
@@ -0,0 +1,232 @@
+    public static final String HEARING_FURTHER_EVIDENCE_DOCUMENTS_COLLECTION_KEY = "hearingFurtherEvidenceDocuments";
+    public static final String C2_SUPPORTING_DOCUMENTS_COLLECTION = "c2SupportingDocuments";
+    public static final String MANAGE_DOCUMENTS_HEARING_LIST_KEY = "manageDocumentsHearingList";
+    public static final String SUPPORTING_C2_LIST_KEY = "manageDocumentsSupportingC2List";
+    public static final String MANAGE_DOCUMENTS_HEARING_LABEL_KEY = "manageDocumentsHearingLabel";
+    public static final String SUPPORTING_C2_LABEL = "manageDocumentsSupportingC2Label";
+    public static final String MANAGE_DOCUMENT_KEY = "manageDocument";
+
+    public Map<String, Object> initialiseManageDocumentEvent(CaseData caseData) {
+        Map<String, Object> listAndLabel = new HashMap<>();
+        String hasHearings;
+
+        if (caseData.getHearingDetails() != null && !caseData.getHearingDetails().isEmpty()) {
+            listAndLabel.put(MANAGE_DOCUMENTS_HEARING_LIST_KEY, caseData.buildDynamicHearingList());
+            hasHearings = YES.getValue();
+        } else {
+            hasHearings = NO.getValue();
+        }
+
+        ManageDocument manageDocument = ManageDocument.builder().hasHearings(hasHearings).build();
+
+        if (caseData.hasC2DocumentBundle()) {
+            listAndLabel.put(SUPPORTING_C2_LIST_KEY, caseData.buildC2DocumentDynamicList());
+        }
+
+        listAndLabel.put(MANAGE_DOCUMENT_KEY, manageDocument);
+
+        return listAndLabel;
+    }
+
+    public Map<String, Object> initialiseHearingListAndLabel(CaseData caseData) {
+        Map<String, Object> listAndLabel = new HashMap<>();
+
+        if (caseData.getManageDocument().isDocumentRelatedToHearing()) {
+            UUID selectedHearingCode = getDynamicListValueCode(caseData.getManageDocumentsHearingList(), mapper);
+            HearingBooking hearingBooking = getHearingBookingByUUID(caseData.getHearingDetails(), selectedHearingCode);
+
+            listAndLabel.put(MANAGE_DOCUMENTS_HEARING_LABEL_KEY, hearingBooking.toLabel(DATE));
+            listAndLabel.put(MANAGE_DOCUMENTS_HEARING_LIST_KEY, caseData.buildDynamicHearingList(selectedHearingCode));
+        }
+
+        return listAndLabel;
+    }
+
+    public Map<String, Object> initialiseC2DocumentListAndLabel(CaseData caseData) {
+        Map<String, Object> listAndLabel = new HashMap<>();
+
+        UUID selectedC2DocumentCode = getDynamicListValueCode(caseData.getManageDocumentsSupportingC2List(), mapper);
+        List<Element<C2DocumentBundle>> c2DocumentBundle = caseData.getC2DocumentBundle();
+
+        IntStream.range(0, c2DocumentBundle.size())
+            .filter(index -> c2DocumentBundle.get(index).getId().equals(selectedC2DocumentCode))
+            .findFirst()
+            .ifPresent(index -> listAndLabel.put(SUPPORTING_C2_LABEL,
+                c2DocumentBundle.get(index).getValue().toLabel(index + 1)));
```

Comment:
I am old but old for is bester in that scenario..
```suggestion
        for (int i = 0; i < c2DocumentBundle.size(); i++) {
            if (c2DocumentBundle.get(i).getId().equals(selectedC2DocumentCode)) {
                listAndLabel.put(SUPPORTING_C2_LABEL, c2DocumentBundle.get(i).getValue().toLabel(i + 1));
                break;
            }
        }
```